pkg/util/file_indexer.go

```diff
@@ -154,32 +154,32 @@ func DeleteIndexFile(directory string) error {
 // after the walk, it stores the list of walked files with some information in a odo index file in the .odo folder
 // The filemap stores the values as "relative filepath" => FileData but it the filesChanged and filesDeleted are absolute paths
 // to the files
-func RunIndexer(directory string, ignoreRules []string) (filesChanged []string, filesDeleted []string, err error) {
+func RunIndexer(directory string, ignoreRules []string) (filesChanged []string, filesDeleted []string, newFileMap map[string]FileData, resolvedPath string, err error) {
```

Comment:
Please update the comment for the function to include a description of what the two new returned values are.

The function now returns 5 values it might be worth to consider if we really need all of them, or maybe converting it to a struct.

Reply:
oops , missed this , yes, need further polishment